examples/psf_deconvolution.py

```diff
@@ -86,50 +89,63 @@
+# Let's compare the convolved and deconvolved images. We will now zoom in on some
+# loop structures that extend off the limb to examine the differences between
+# our image before and after deconvolution.
+#
 # The differences become a bit more obvious when we zoom in. Note that the
 # deconvolution has the effect of "deblurring" the image.
 
-left_corner = 500 * u.arcsec, -600 * u.arcsec
-right_corner = 1000 * u.arcsec, -100 * u.arcsec
-
-aia_map_sub = aia_map.submap(
-    bottom_left=SkyCoord(*left_corner, frame=aia_map.coordinate_frame),
-    top_right=SkyCoord(*right_corner, frame=aia_map.coordinate_frame),
-)
-aia_map_deconvolved_sub = aia_map_deconvolved.submap(
-    bottom_left=SkyCoord(*left_corner, frame=aia_map_deconvolved.coordinate_frame),
-    top_right=SkyCoord(*right_corner, frame=aia_map_deconvolved.coordinate_frame),
-)
-
-fig = plt.figure(figsize=(12, 8))
+bottom_left = SkyCoord(750, -375, unit="arcsec", frame=aia_map.coordinate_frame)
+fov = {"width": 400 * u.arcsec, "height": 400 * u.arcsec}
+
+aia_map_sub = aia_map.submap(bottom_left, **fov)
+aia_map_deconvolved_sub = aia_map_deconvolved.submap(bottom_left, **fov)
+
+fig = plt.figure(figsize=(10, 6))
+ax1 = fig.add_subplot(121, projection=aia_map_sub)
+aia_map_sub.plot(axes=ax1, title="Before Deconvolution")
+ax2 = fig.add_subplot(122, projection=aia_map_deconvolved_sub)
+aia_map_deconvolved_sub.plot(axes=ax2, title="After Deconvolution")
+# Just remove the ticks and labels
+lon = ax2.coords[0]
+lat = ax2.coords[1]
+lon.set_ticks_visible(False)
+lon.set_ticklabel_visible(False)
+lat.set_ticks_visible(False)
+lat.set_ticklabel_visible(False)
+lon.set_axislabel("")
+lat.set_axislabel("")
+fig.tight_layout()
 
-ax = fig.add_subplot(121, projection=aia_map_sub)
-aia_map_sub.plot(axes=ax, norm=norm)
-ax.set_title("Normal")
+###############################################################################
+# Note that comparing the images on the left and right, deconvolving
+# with the PSF removes the appearance of the diffraction pattern near the
+# bright loop top. This also brings out more detail in the strands of the
+# post-flare arcade.
+#
+# We can see this more clearly by plotting the intensity of the original
+# and deconvolved images across the top of the loop top.
 
-ax = fig.add_subplot(122, projection=aia_map_deconvolved_sub)
-aia_map_deconvolved_sub.plot(axes=ax, annotate=False, norm=norm)
-ax.set_title("Deconvolved")
-ax.coords[0].set_axislabel(" ")
-ax.coords[1].set_axislabel(" ")
-ax.coords[1].set_ticklabel_visible(visible=False)
+x_locations = np.linspace(
+    aia_map_deconvolved_sub.dimensions.x.value * 0.55, aia_map_deconvolved_sub.dimensions.x.value * 0.7
+)
+y_locations = 0.59 * aia_map_deconvolved_sub.dimensions.y.value * np.ones(x_locations.shape)
+slit_locations = np.s_[
+    np.round(y_locations[0]).astype(int), np.round(x_locations[0]).astype(int) : np.round(x_locations[-1]).astype(int)
+]
```

Comment:
This can now all be replaced with https://docs.sunpy.org/en/stable/generated/api/sunpy.map.pixelate_coord_path.html#sunpy.map.pixelate_coord_path

Reply:
Updated.